atom/browser/api/lib/web-contents.coffee

```diff
@@ -34,6 +34,12 @@ module.exports.wrap = (webContents) ->
     do (name, method) ->
       webContents[name] = -> method.apply controller, arguments
 
+  # Service-worker utilities.
+  webContents.hasServiceWorker = (callback) ->
+    @_hasSW @getUrl(), callback
```

Comment:
I went with `web_contents()->GetLastCommittedURL()` as its safe to use now, since there will be only one entry (current loaded), as we are clearing the history list before load.